src/Stratis.Bitcoin.Features.Apps/AppsStore.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Features.Apps.Interfaces;
+using Stratis.Bitcoin.Utilities.Extensions;
+
+namespace Stratis.Bitcoin.Features.Apps
+{
+    public class AppsStore : IAppsStore
+    {
+        private readonly ILogger logger;
+        private readonly List<IStratisApp> applications = new List<IStratisApp>();
+        private readonly IAppsFileService appsFileService;
+        private readonly IStratisAppFactory appFactory;
+
+        public AppsStore(ILoggerFactory loggerFactory, IAppsFileService appsFileService, IStratisAppFactory appFactory)
+        {
+            this.appsFileService = appsFileService;
+            this.appFactory = appFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.Load();
+        }
+
+        public IEnumerable<IStratisApp> Applications => this.applications;
+
+        private void Load()
+        {
+            IEnumerable<FileInfo> configs = this.appsFileService.GetStratisAppConfigFileInfos();
+            var apps = configs.Select(this.CreateApp).Where(x => x != null).ToList();
+
+            if (apps.IsEmpty())
+            {
+                this.logger.LogWarning(
```

Comment:
As a matter of policy we don't use string interpolation in logs (for performance).
You can do `this.logger.LogWarning("No Stratis applications found at or below {0}", "this.appsFileService.StratisAppsFolderPath");`.

Reply:
OK